src/components/creator/schema.tsx

```diff
@@ -229,106 +120,64 @@ export function makeSchema(chrome: ChromeAPI): Schema {
-            component: componentTypes.TEXTAREA,
-            name: NAME_PANEL_INTRODUCTION,
-            label: 'Introduction (Markdown)',
-            resizeOrientation: 'vertical',
-          },
-          {
-            component: componentTypes.FIELD_ARRAY,
-            name: NAME_PREREQUISITES,
-            label: 'Prerequisites',
-            noItemsMessage: 'No prerequisites have been added.',
-            fields: [
-              {
-                component: componentTypes.TEXT_FIELD,
-                label: 'Prerequisite',
-              },
-            ],
-          },
-          {
-            component: componentTypes.FIELD_ARRAY,
-            name: NAME_TASK_TITLES,
-            label: 'Tasks',
-            minItems: 1,
-            maxItems: MAX_TASKS,
-            noItemsMessage: 'No tasks have been added.',
-            initialValue: [''],
-            fields: [
-              {
-                component: componentTypes.TEXT_FIELD,
-                label: 'Title',
-              },
-            ],
-          },
-        ],
-        nextStep: taskStepName(0),
-      },
+      makeKindStep(),
+      ...ALL_ITEM_KINDS.map((kind) =>
+        makeDetailsStep({ kind, bundles, downloadStep: STEP_DOWNLOAD })
+      ),
+      ...ALL_ITEM_KINDS.filter((kind) => metaForKind(kind).hasTasks).map(
+        (kind) =>
+          makePanelOverviewStep({
+            kind,
+            panelStepTitle: STEP_TITLE_PANEL_PARENT,
+          })
+      ),
       ...taskSteps,
-      {
-        name: STEP_DOWNLOAD,
-        title: 'Download files',
-        fields: [
-          {
-            component: 'lr-download-files',
-            name: 'internal-download',
-          },
-        ],
-      },
+      makeDownloadStep(),
     ],
   };
 
-  const schema = {
-    fields: [wizardProps],
-  };
-
-  for (const step of schema.fields) {
-    if (step.component === componentTypes.WIZARD) {
-      for (const page of step.fields) {
-        // Add an lr-wizard-spy component to all wizard steps. It must be here (rather
-        // than at the top level of the schema) so that it is inside the WizardContext.
-        page.fields.push({
+  const wizardProps = {
+    ...rawWizardProps,
+    fields: rawWizardProps.fields.map((rawPage) => {
+      const page: typeof rawPage & {
+        buttonLabels?: { [key: string]: string };
+      } = { ...rawPage };
+
+      // Add an lr-wizard-spy component to all wizard steps. It must be here (rather
+      // than at the top level of the schema) so that it is inside the WizardContext.
+      page.fields = [
+        ...page.fields,
+        {
           component: 'lr-wizard-spy',
           name: `internal-wizard-spies.${page.name}`,
-        });
+        },
+      ];
 
-        // Use custom buttons for each step.
-        if (page.buttons === undefined) {
-          page.buttons = CustomButtons;
+      // Use custom buttons for each step.
+      if (page.buttons === undefined) {
+        page.buttons = CustomButtons;
```

Comment:
Seems like the buttons are in wrong order:
![screenshot-stage_foo_redhat_com_1337-2024_07_17-10_19_00](https://github.com/user-attachments/assets/b3070cfa-f6f9-489f-93c6-371c00775de6)